apps/emqx_auth_mnesia/src/emqx_acl_mnesia_cli.erl

```diff
@@ -233,3 +239,13 @@ print_acl({all, Topic, Action, Access, _}) ->
         "Acl($all topic = ~p action = ~p access = ~p)~n",
         [Topic, Action, Access]
      ).
+
+maybe_delete_shadowed_records(_, []) ->
+    ok;
+maybe_delete_shadowed_records(Action1, [Rec = #emqx_acl{action = Action2} | Rest]) ->
+    if Action1 =:= Action2 orelse Action1 =:= pubsub ->
+            ok = mnesia:delete_object(Rec);
```

Comment:
Hi, @k32 
If I have `{{username, usr1}, topic1}, deny, pubsub`, then insert a new rule `{{username, usr1}, topic1}, allow, pub`, then `{{username, usr1}, topic1}, deny, pubsub` will be deleted, this will have side effects

Reply:
Good catch, thanks

Reply:
@Rory-Z Maybe the best course of action is to replace `pubsub` with two separate `pub` and `sub` records in the database? But keep it in the presentation layer.

Reply:
good idea.
I'm not sure why we should keep it in the presentation layer.

Reply:
Sounds good to me. I kept the ability to specify `pubsub` permission for backward compatibility, but it doesn't show up anywhere otherwise.